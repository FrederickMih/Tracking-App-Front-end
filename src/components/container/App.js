import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import 'bootstrap/dist/css/bootstrap.min.css';
import LoginUser from '../presentation/LoginUser';
import { getMeasurements } from '../../actions';
import Router from '../../router/Router';
import Header from './Header';
// import LogoutUser from '../presentation/LogoutUser';
import '../../styles/App.css';

const App = () => {
  const dispatch = useDispatch();
  // get measurements data from the endpoint
  useEffect(() => {
    dispatch(getMeasurements());
  }, [dispatch]);
  // Load user data from state
  const user = useSelector((state) => state.user);

  let app = <LoginUser />;
  if (user.loggedIn) {
    app = (
      <>
        <Header />
        <Router />
      </>
    );
  }

  return <div className="App">{app}</div>;
};

export default App;
