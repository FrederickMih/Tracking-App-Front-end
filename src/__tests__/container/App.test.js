import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import App from '../../components/container/App';
import store from '../../reducers/index';

afterEach(cleanup);

it('should render the App component snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(
    asFragment(
      <Provider store={store}>
        <App />
      </Provider>
    )
  ).toMatchSnapshot();
});

it('Displays username field', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  const element = screen.getByPlaceholderText(/Username/i);
  expect(element).toBeInTheDocument();
});

it('Displays Email field', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  const element = screen.getByPlaceholderText(/Email/i);
  expect(element).toBeInTheDocument();
});
