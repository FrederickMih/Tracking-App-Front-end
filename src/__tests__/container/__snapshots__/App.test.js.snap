// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the App component snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="login__page"
    >
      <form
        class="login__form"
      >
        <h1>
          Login Here
        </h1>
        <input
          id="username"
          placeholder="Username"
          type="username"
        />
        <input
          id="email"
          placeholder="Email"
          type="email"
        />
        <button
          class="submit__button"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
